tools/automation/twiml-generator/generator.py

```diff
@@ -24,12 +26,82 @@ def generate_code_sample_filepath(twiml_filepath, language):
     return twiml_filepath.parent.parent / (twiml_filepath.name[:-3] + version + language_spec['extension'])
 
 
+def test(code_generator):
```

Comment:
It's not a duplicated effort, the comparison code was already in the twiml-generator. Only the way to run it for Java and C# was added here. Ultimately the library should also provide those feature, but it was easier to place here for now.

The CI must test all the snippets to avoid any commit that can interfere with those snippets but the generator must not output a code that is wrong without warning. Having immediate returns of compilation errors or differences between the input and output XML are a big help when updating the generator. Most of the code here is not cleaned yet and have more to do with the terminal output than the actual running / evaluating the output.